lib/image-service-url.js

```diff
@@ -6,20 +6,103 @@ module.exports = class ImageServiceUrl {
 
 	constructor(urlString) {
 		const parsedUrl = url.parse(urlString, true);
-
 		this.source = decodeURIComponent(parsedUrl.pathname.replace('/v1/images/raw/', ''));
-		this.width = numberOrUndefined(parsedUrl.query.width);
-		this.height = numberOrUndefined(parsedUrl.query.height);
-		this.dpr = numberOrUndefined(parsedUrl.query.dpr);
-		this.fit = parsedUrl.query.fit || 'cover';
-		this.format = parsedUrl.query.format;
+
+		this.setWidth(parsedUrl.query.width);
+		this.setHeight(parsedUrl.query.height);
+
+		this.setDpr(parsedUrl.query.dpr);
+		this.setFit(parsedUrl.query.fit);
+		this.setFormat(parsedUrl.query.format);
+
+		this.setQuality(parsedUrl.query.quality);
 	}
 
-};
+	setWidth(value) {
+		this._setNumericProperty('width', value);
+	}
 
-function numberOrUndefined(value) {
-	if (value === undefined) {
-		return value;
+	setHeight(value) {
+		this._setNumericProperty('height', value);
 	}
-	return parseInt(value, 10);
-}
+
+	setDpr(value) {
+		this._setNumericProperty('dpr', value);
+	}
+
+	setFit(value = 'cover') {
+		this._setEnumerableProperty('fit', ImageServiceUrl.validFitValues, value);
+	}
+
+	setFormat(value = 'auto') {
+		this._setEnumerableProperty('format', ImageServiceUrl.validFormatValues, value);
+		if (value === 'auto') {
+			if (this.qualityName === 'lossless') {
+				return this.format = 'png';
+			}
+			this.format = 'jpg';
+		}
+	}
+
+	setQuality(value = 'medium') {
+		this._setEnumerableProperty('quality', ImageServiceUrl.validQualityValues, value);
+		this.qualityName = this.quality;
+		this.quality = ImageServiceUrl.qualityNameValueMap[this.qualityName];
+	}
+
+	_setNumericProperty(property, value) {
+		if (value === undefined) {
+			return this[property] = value;
+		}
+		if (typeof value !== 'string' && typeof value !== 'number') {
+			throw new Error(`Invalid ${property} parameter`);
+		}
+		value = Number(value);
```

Comment:
OK gonna merge and will revise when we come up with a standard :) maybe a good discussion point in the next Origami meeting? We don't even use half of ESLint's capabilities